packages/lib/src/components/UIElement.tsx

```diff
@@ -71,46 +80,49 @@ export class UIElement<P extends UIElementProps = any> extends BaseElement<P> {
     /**
      * Submit payment method data. If the form is not valid, it will trigger validation.
      */
-    submit(): void {
+    public submit(): void {
         if (!this.isValid) {
             this.showValidation();
-            return null;
+            return;
         }
 
         this.onSubmit();
     }
 
-    showValidation(): this {
+    public showValidation(): this {
         if (this.componentRef && this.componentRef.showValidation) this.componentRef.showValidation();
         return this;
     }
 
-    setStatus(status, props?): this {
-        if (this.componentRef && this.componentRef.setStatus) this.componentRef.setStatus(status, props);
+    public setStatus(status: UIElementStatus, props?): this {
+        if (this.props.isDropin && this.elementRef?.setStatus) {
+            this.elementRef.setStatus(status, props);
+        } else if (this.componentRef && this.componentRef.setStatus) {
+            this.componentRef.setStatus(status, props);
+        }
         return this;
     }
 
-    submitPayment(data): Promise<void> {
-        this.setStatus('loading');
```

Comment:
So is this setting of status no longer needed?

Reply:
The status is set here before calling `submitPayment`, in this part:

https://github.com/Adyen/adyen-web/blob/b397d48b8850e42172f1904cd567bb8dd5968917/packages/lib/src/components/UIElement.tsx#L51-L53

Reply:
ok